sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/LookupCatalog.scala

```diff
@@ -62,6 +62,9 @@ trait LookupCatalog extends Logging {
     try {
       Some(lookupCatalog(SESSION_CATALOG_NAME))
     } catch {
+      case _: CatalogNotFoundException =>
+        logWarning("Session catalog is not defined")
+        None
```

Comment:
@jzhuge . `CatalogNotFoundException` is already handled at by `case NonFatal`, isn't it?